src/uwtools/resources/rocoto.jsonschema

```diff
@@ -0,0 +1,72 @@
+{
+    "definitions": {
+        "dateTimePattern": {
+            "type": "string",
+            "pattern": "^(\\d{12}) (\\d{12}) (\\d{2}):(\\d{2}):(\\d{2}):(\\d{2})$"
+        }
+    },
+    "type": "object",
+    "properties": {
+        "workflow": {
+            "type": "object",
+            "properties": {
+                "attrs": {
+                    "type": "object",
+                    "properties": {
+                        "realtime": {
+                            "type": "string",
+                            "enum": [
+                                "true",
+                                "false",
+                                "T",
+                                "F",
+                                "t",
+                                "f",
+                                "True",
+                                "False",
+                                "FALSE",
+                                "TRUE"
+                            ]
+                        },
+                        "scheduler": {
+                            "type": "string",
+                            "enum": [
+                                "sge",
+                                "lsf",
+                                "lsfcray",
+                                "ll",
+                                "torque",
+                                "moabtorque",
+                                "moab",
+                                "pbspro",
+                                "slurm",
+                                "cobalt"
+                            ]
+                        },
+                        "cycylethrottle": {
+                            "type": "integer",
+                            "minimum": 0
+                        },
+                        "taskthrottle": {
+                            "type": "integer",
+                            "minimum": 0
+                        }
+                    },
+                    "required": ["realtime", "scheduler"]
+                },
+                "cycledefs": {
+                    "$ref": "#/definitions/dateTimePattern"
+                },
+                "log": {
+                    "type": "string",
+                    "format": "uri"
+                },
+                "entities": {
+                    "type": "object"
+                }
+            },
+            "required": ["attrs", "cycledefs", "log"]
```

Comment:
I let entities off the `required` list since they are not required to be supplied by the user

Reply:
Would validation fail a YAML block with `entities` if they were not at least included as _optional_ items of the schema? How does JSON Schema treat extraneous items that are not part of the schema?

Reply:
I was able to add in that example `hello_workflow_tags` that includes an `entities` block and it passes. It is included in the schema as `optional`. Meaning if it's not in the `required` section it is `optional`. 

I've decided to make a `hello_workflow_invalid.yaml` that tests what happens if an item that is not part of the schema and it should fail. After going through the tests for each of the `tags`: `attrs`, `cycledef`, `log`, `entities`. All fail as if an input is not as expected, EXCEPT `cycledef`.

I added a test to see if it would catch if the format of the `cycledef` did not match the regex. It does not catch that correctly. It should pass the `assert not` with true in the tests.

Not sure exactly why it is passing the regex match.

Reply:
Tried to play around with the idea of making `groupname` and `$ref` required using `"required": ["groupname", "$ref"]`

Reply:
@christinaholtNOAA